src/tennissimulator/menu/Menu.java

```diff
@@ -100,61 +107,55 @@ private Gender resolveGender(String gender) {
 
     private int promptUserForRating(Organization organization) {
         System.out.print("Enter a " + organization.toString() + " rating: ");
+
         return scanner.nextInt();
     }
 
     private String promptUserForName() {
         System.out.print("Enter a full player name: ");
+
         return scanner.nextLine();
     }
 
     private String promptUserForGender() {
         System.out.print("Enter a gender (man/woman): ");
+
         return scanner.nextLine();
     }
 
-    private void selectPlayersForMatchSimulation() {
-        playersDatabase.print();
-        if (playersDatabase.getPlayers().isEmpty()) {
-            return;
-        }
+    private String promptUserForId() {
+        System.out.println("\nChoose player ID: ");
 
-        Player firstPlayer = resolvePlayer(promptUserForName());
-        if (!isValidPlayer(firstPlayer)) {
-            return;
-        }
+        return scanner.nextLine();
+    }
 
-        Player secondPlayer = resolvePlayer(promptUserForName());
-        if (!isValidPlayer(secondPlayer)) {
+    private void selectPlayersForMatchSimulation() throws RuntimeException {
+        playersDatabase.print();
+        if (playersDatabase.getPlayers().isEmpty()) {
             return;
         }
 
-        match = new Match(firstPlayer, secondPlayer);
-        System.out.println("\n" + firstPlayer.getName() + " is playing against "
-                + secondPlayer.getName() + "\n");
-    }
-
-    private boolean isValidPlayer(Player player) {
-        if (player == null) {
-            System.out.println("Can`t find player in database. Please enter valid name.");
-            return false;
+        Optional<Player> firstPlayer = playersDatabase.findPlayer(Integer.parseInt(promptUserForId()));
+        Optional<Player> secondPlayer = playersDatabase.findPlayer(Integer.parseInt(promptUserForId()));
+        if (firstPlayer.isEmpty() || secondPlayer.isEmpty()) {
+            throw new RuntimeException();
```

Comment:
You didn't introduce a new exception. Runtime is already existing Java exception and it does not say anything. Imagine it would be thrown, in such case it wouldn't  be clear from logs what was the reason of it because you didn't add any message.

1. It is not recommended throwing generic RuntimeException because if application has 20 places throwing RuntimeExceptions then you will have hard time debugging it. Instead you should introduce your own exception which extends RuntimeException and is self explaining.  

2. exception should have message which adds more details to it. Again, exception without message does not help in debugging.
 
3.  You check whether firstPlayer isEmpty only after entering second player. You should fail fast (https://en.wikipedia.org/wiki/Fail-fast) and not allow entering second player if first is invalid.

Reply:
introduced custom `InvalidPlayerIdException`